Felinid Thievery

WarMechanic opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
This PR updates Felinids with a soft rework to their old passive thieving gloves ability. Now, Felinids have instead learned to see hidden items when stripping other mobs (i.e. items in pockets) and strip items faster in general (100% -> 66.67%). Admin Observers get this ability too.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
Since the removal of passive thieving gloves (see https://github.com/DeltaV-Station/Delta-v/pull/666), Felinids cannot so much be called pickpockets anymore since they get little to their name that benefits such. Allowing Felinids to see into players' pockets and increasing stripping speed are parts of my suggested changes to work the pickpocket identity back into the Felinid kit while having implications for both antags and non-antags.


> Weren't passive thieving gloves removed for good reason? / Won't this change promote shittery? / Any other variation thereof.

Passive stealth was indeed removed for a very sensible and good reason, but I would like to make the distinction that passive stealth enabled uncontested shittery since Felinids could simply have gloves, magboots or even belts disappear with little retaliation and RP value by extension, this comparison honestly isn't fair. This PR addresses this concern by leaving out stealth and only introducing pocket sight and strip speed buffs. I tried to implement this new set of changes with non/antag viability in mind. While old Felinids' passive thieving gloves let you skip on 4tc to rob people, this new iteration doesn't hide doafter which allows for retaliation. Any attempts made to steal, irrespective of success can and will go noticed by players (unless thieving gloves are used). But you can even combine thieves gloves with Felinid strip speed as an antag Felinid for a fastest thief gimmick (has similar energy to species specific traitor items). Pocket sight can give any player useful information and faster stripping has implications in both the Security department and everyday stealing. 

At the end of the day, Felinids could likely be used by 'shitters' to steal an item or two. But remember that stripping is an action largely restricted to roleplay and non-combat scenarios. As such, this rebalance would not affect parts of the game an Oni's stat stick with accuracy penalty would (the average greytider will not use a gun, and benefits from increased crowbar/baseball damage). This buff will likely balance itself out over time to rising awareness and an anticipated 'Felinid = thief' stigma. Subversions of this stigma provides opportunity for roleplay.

> Why pocket sight?

Felinid pocket sight grants them a new way to comb information that's largely befitting of their roguish archetype. While in the past the same results could have been achieved with passive stealth, this ability creates agency regarding the value of an item and whether to actually attempt stealing it in the first place (especially since you no longer have passive stealth). Having information unique to a species also creates opportunity for 'boy who cried wolf' situations wherein a set of people decide on the merit of a Felinid's argument: "i saw an e-saber in his pocket!!!", as well as providing reason for Antags to distrust Felinids (nobody should trust Felinids).

> Why pocket sight AND strip speed buff?

While the PR could be left with just pocket sight, pockets only pertain to two item slots where one of them likely holds an emergency oxygen tank anyway. But most importantly pocket sight is likely to go unnoticed by a majority of players that do not play Felinids or look at the github / changelogs, since pocket sight rejects an established notion that you cannot see into another players' pocket. It almost feels metagamey, like an afterthought to mention it. Adding a strip speed buff is very much complementary to the pocket sight addition since it provides diegetic ground for players to acknowledge Felinids' pickpocket potential. While pocket sight only directly affects the user, faster stealing can be witnessed by all players. As for the numbers, 66.67% strip time brings down pocket/radio/shoes/hand strip times from (3 -> 2) seconds and backpack (6 -> 4) seconds. 2 seconds I've observed to be the reaction time of most players, so Felinids will be able to nip items more consistently. I chose multiplicative strip time as faster belt stripping has better implications for Security when searching for contraband. If one were to have their belt stolen, it is likely their fault for not paying attention and would have had it stolen anyway.

I talk to myself here regarding balance choices: https://discord.com/channels/968983104247185448/1228169033547776152

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Content.Client/.../StrippableBoundUserInterface.cs checks for ThievingComponent and its IgnoreStripHidden variable on the local entity before obscuring hidden items. Because this is client-sided I do not know if this is the safest implementation of this feature.

StripTimeMultiplier variable was also added to Thieving, which simply updates a preexisting multiplier argument that's passed to an event.

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

![image](https://github.com/DeltaV-Station/Delta-v/assets/69510347/341e092a-d86a-4acc-b395-908db8f98ee4)

https://youtu.be/cAVxXPUNImI?si=fBuKBQHGrzzJn03f

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- tweak: Added the IgnoreStripHidden variable to ThievingComponent, which allows entities to look into the pockets of mobs it strips.
- tweak: Added the StripTimeMultiplier variable to ThievingComponent, which multiplies strip time.
- add: Felinids and Admin Observers are now able to identify items in pockets of other players. Felinids also gain a faster strip speed.

WarMechanic committed:
Strip UI can now check for Thieving Component to see hidden items

WarMechanic committed:
Admin Observers can now see hidden items in strip menu

WarMechanic committed:
Felinid pickpocketing buffs

WarMechanic committed:
Revert "Felinid pickpocketing buffs"

This reverts commit cfc6d807197465ac3f82f815e89505acbdfec275.

WarMechanic committed:
Revert "Admin Observers can now see hidden items in strip menu"

This reverts commit 496188729fba60e15a1b05d82feab641051e2607.

WarMechanic commented on Content.Shared/Strip/Components/ThievingComponent.cs:
I don't know what SixLabors.ImageSharp is, or how it got there. Allow me to remove it.

WarMechanic committed:
Removed mysterious line of code

WarMechanic committed:
Merge branch 'master' into thieving

deltanedas commented on Content.Shared/Strip/Components/ThievingComponent.cs:
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("stripTimeReduction")]
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeReduction")]
```
dont touch this

deltanedas commented on Content.Shared/Strip/Components/ThievingComponent.cs:
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("stealthy")]
+    [ViewVariables(VVAccess.ReadWrite), DataField("stealthy")]
```
or this

deltanedas commented on Content.Shared/Strip/Components/ThievingComponent.cs:
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
+    /// <summary>
+    ///  Should the user be able to see hidden items? (i.e pockets)
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("ignoreStripHidden")]
```
replace this line with [DataField] and add deltav marker comment

WarMechanic replied:
What is a deltav marker comment? Is it literally a comment that signifies that the line of code pertains to DeltaV?

deltanedas replied:
yes

deltanedas commented on Content.Shared/Strip/Components/ThievingComponent.cs:
```diff
@@ -9,14 +9,21 @@ public sealed partial class ThievingComponent : Component
     public TimeSpan StripTimeReduction = TimeSpan.FromSeconds(0.5f);
 
+    [ViewVariables(VVAccess.ReadWrite), DataField("stripTimeMultiplier")]
+    public float StripTimeMultiplier = 1f;
```
this is unused remove

WarMechanic replied:
StripTimeMultiplier I used for Felinid buff. Should I remove it and readd it in its own PR?

deltanedas replied:
there is no felenid buff you reverted that

also felinids having thieving was removed because felenids were ultimate shitter/syndie species

WarMechanic replied:
Thieving was removed I presume because items vanishing from inventories without a trace would be frustrating which I agree is sensible. However my buff would function as a soft reintroduction of a pickpocket passive that has more implications for non-antags. I have a discussion here: https://discord.com/channels/968983104247185448/1228169033547776152

deltanedas commented on Content.Client/Inventory/StrippableBoundUserInterface.cs:
```diff
@@ -31,6 +32,7 @@ namespace Content.Client.Inventory
     public sealed class StrippableBoundUserInterface : BoundUserInterface
     {
         [Dependency] private readonly IUserInterfaceManager _ui = default!;
+        [Dependency] internal readonly IPlayerManager PlayerManager = default!;
```
private not internal, nothing else uses it

also deltav marker here and on the usage line

WarMechanic committed:
Response to review

WarMechanic committed:
Merge branch 'thieving' of https://github.com/WarMechanic/Delta-v into thieving

Merged new commits into branch to push

WarMechanic committed:
Removed more stuff unrelated to topic

deltanedas commented on Content.Shared/Strip/Components/ThievingComponent.cs:
```diff
@@ -19,4 +19,11 @@ public sealed partial class ThievingComponent : Component
+    ///  Should the user be able to see hidden items? (i.e pockets)
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("ignoreStripHidden")] //DeltaV
```
again just [DataField] its automatically rw and uses camelCase variable name

= false can be removed thats the default

also nothing uses it so...

WarMechanic replied:
Can you rephrase this? I don't know what you want me to do

deltanedas replied:
delete those lines and have

```cs
[DataField]
public bool IgnoreStripHidden; // DeltaV
```

then make it actually used by aghost and felenid prototype, but felenids should still show doafter

WarMechanic committed:
Responded to review regarding syntax

WarMechanic committed:
Added IgnoreStripHidden to Felinids and Admin Observers

Axionyxx commented:
Felinid thiefing was removed on purpose

deltanedas commented:
this has no reduction and still shows doafter so seems ok balance wise

WarMechanic commented:
What's wrong with reduction? I believe that reduction while still maintaining doafter while allowing you to successfully take something, wouldn't let you get away with said item for free. Most people respond to pickpocketing over the halfway mark so I thought it would be a fair change.

Multiplicative reduction which I reverted earlier I implemented because multiplicative strip time reduction has use cases in Security to strip belts and bags off criminals to search for contraband faster.

Although at the end of the day I will probably put it in a separate PR so the changes are easily revertible.

WarMechanic commented:
I've decided that there may have been something I misunderstood regarding the PR guidelines on Robust Book since the scope of this PR was originally going to be the tweak by itself.

I would greatly appreciate advice on how to treat this PR going forward.

WarMechanic committed:
Reimplement strip time multiplier with correct syntax

WarMechanic committed:
Added strip time multiplier to Felinid and Admin Observer

WarMechanic commented:
How do I fix the outdated review?

WarMechanic reviewed (commented):
Testing, can this outdated review resolve?

deltanedas reviewed (approved):
i approve of the implementation

WarMechanic commented:
holy shit

rosieposieeee commented:
in my completely one hundred percent unbiased opinion, i think vulps should get this too since they lost thieving gloves as well.

WarMechanic commented:
huh, skub

i never saw any remove thieving from vulps pr but yeah i looked at the first pr and thieving is there. i think pickpocketing fits with felinids because they have a small size, but it should be easy to copy and paste the thieving component from felinid to vulp if you want to pr it.

WarMechanic commented:
how do i fix do not merge, is it a balance concern

WarMechanic commented:
thanks a bunch deltanedas

WarMechanic committed:
Merge branch 'master' of https://github.com/DeltaV-Station/Delta-v into thieving

merged update to pr

WarMechanic committed:
Merge branch 'master' of https://github.com/WarMechanic/Delta-v into thieving

WarMechanic committed:
Merge branch 'master' into thieving

WarMechanic commented:
any word on this? slimes can turn into geras and have minor internal storage now

therealDLondon commented:
personally from a balance standpoint im fine with this, Especially considering other species have been buffed somewhat recently
